### PR TITLE
EL-3727: Change docker e2e test execution

### PR DESCRIPTION
**Pull request by andyreidz**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licenced under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue
https://portal.digitalsafe.net/browse/EL-3727

#### Description of Proposed Changes
User a dedicated directory on the docker host environment for Linux node modules.

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_

### CODE REVIEW

File: scripts/run-ci-local.js [hunk 1]
```diff
@@ -1,60 +1,36 @@
...
-    // Move existing node_modules to a known location
-    renameSync(nodeModulesDir, nodeModulesDevDir);
-    console.log(`Renamed node_modules to ${basename(nodeModulesDevDir)}. You can use this instead of doing \`npm ci\` again after exiting the shell.`);
-} catch (error) {
-    if (error.code !== 'ENOENT') {
-        console.warn(`Couldn't rename node_modules. (The IDE probably has a handle on it.)`);
-    }
-}
+// Ensure the docker node_modules__docker directory exists
+mkdirpSync(nodeModulesDocker);
```
Comment (donna-marijne) [inline scripts/run-ci-local.js]: This should be `nodeModulesDockerDir` (or rename the above variable).

File: scripts/run-ci-local.js [hunk 2]
```diff
@@ -1,60 +1,36 @@
...
-// access environment
-const http_proxy = process.env.HTTP_PROXY;
-const https_proxy = process.env.HTTPS_PROXY;
+if (!https_proxy){
+    console.warn(`An https proxy has not been defined.`);
+}
 
 // produce the docker command string
-const dockerCommand = `docker run --rm -it --memory=4g -e "http_proxy=${http_proxy}" -e "https_proxy=${https_proxy}" -v "${cwd}:/wd" -w /wd --entrypoint /bin/bash uxaspects/buildenv:latest`;
+const dockerCommand = `docker run --rm -it --memory=4g -e "http_proxy=${http_proxy}" -e "https_proxy=${https_proxy}" -v "${cwd}:/wd" -v "${nodeModulesDocker}:/wd/node_modules" -w /wd --entrypoint /bin/bash ${dockerImage}`;
```
Comment (donna-marijne) [inline scripts/run-ci-local.js]: `nodeModulesDockerDir` should be used here as well.

File: scripts/run-ci-local.js [hunk 3]
```diff
@@ -1,60 +1,36 @@
...
 const { join, basename } = require('path');
+const { mkdirpSync } = require('fs-extra');
 
 const cwd = process.cwd();
 
 const nodeModulesDir = join(cwd, 'node_modules');
-const nodeModulesDevDir = join(cwd, '.node_modules__dev');
 const nodeModulesDockerDir = join(cwd, '.node_modules__docker');
+//const dockerImage = 'uxaspects/buildenv:latest';
+const dockerImage = 'cafinternal/prereleases:buildenv-1.3.0-EL-3727-SNAPSHOT';
```
[the remark this answers is not in the document]
Comment (andyreidz) [reply scripts/run-ci-local.js]: It's left like this until we release the build image

File: scripts/run-ci-local.js [hunk 4]
```diff
@@ -1,60 +1,36 @@
...
-    renameSync(nodeModulesDockerDir, nodeModulesDir);
-    console.log(`Restored node_modules from ${basename(nodeModulesDockerDir)}. Running \`npm ci\` may not be necessary.`);
-} catch (error) {
-    console.log('Run `npm ci` in the docker shell to obtain Linux 64-bit dependencies.');
+// access environment
+const http_proxy = process.env.HTTP_PROXY || process.env.http_proxy;
+const https_proxy = process.env.HTTPS_PROXY || process.env.https_proxy;
+
+if (!http_proxy){
+    console.warn(`An http proxy has not been defined.`);
```
[the remark this answers is not in the document]
Comment (andyreidz) [reply scripts/run-ci-local.js]: Didn't want to suggest that we only looked at the capitalized env. Other platforms has case sensitive envs.
https://github.com/nodejs/node/issues/9157

File: scripts/run-ci-local.js (hunk 1 above)
Comment (andyreidz) [reply scripts/run-ci-local.js]: Argh, still getting used to VS Code and it not auto saving :)

File: scripts/run-ci-local.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (donna-marijne) [reply scripts/run-ci-local.js]: How about 'Neither HTTP_PROXY nor http_proxy has not been defined in the host environment.'?

File: scripts/run-ci-local.js (hunk 1 above)
Comment (donna-marijne) [reply scripts/run-ci-local.js]: There is an auto-save setting, although I can't vouch for it:
image

File: scripts/run-ci-local.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (andyreidz) [reply scripts/run-ci-local.js]: I've assigned the PR to myself. I won't merge until the image is released.

[the remark this answers is not in the document]
Comment (donna-marijne) [reply scripts/run-ci-local.js]: No worries, FYI there is a dropdown when creating a PR which lets you choose to make a draft PR which can't be merged until you mark it ready.

File: scripts/run-ci-local.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (andyreidz) [reply scripts/run-ci-local.js]: You mean
How about 'Neither HTTP_PROXY nor http_proxy have been defined in the host environment.'?

:)

**Comment by donna-marijne** [comment]
This really should not have been merged with temporary code in place.

**Pull request merged**